### PR TITLE
Mike | Added player header info. Still a work in progress css wise an…

### PR DESCRIPTION
**Pull request by mike-robertson**
…d tooltips

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   },
   "extends": "airbnb",
   "rules": {
-    "react/prop-types": 1,
+    "react/prop-types": 0,
```
Comment (howardchung) [inline .eslintrc.json]: too many warnings?

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
 node_modules
 .idea
 npm-debug.log
-build
\ No newline at end of file
+build
```
Comment (howardchung) [inline .gitignore]: remove the built file

File: components/Player/Player.jsx [hunk 3]
```diff
@@ -1,10 +1,58 @@
...
+      this.props.getPlayer(nextProps.playerId);
+    }
+  }
+
+  render() {
+    return <Player {...this.props} />;
+  }
+}
+
+export default withRouter(connect(mapStateToProps, mapDispatchToProps)(RequestLayer));
```
Comment (howardchung) [inline components/Player/Player.jsx]: I'm not sure you need this withRouter.  See what I did in the Match component, you can read the Router state directly from the component.

File: components/Player/PlayerName.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
```
Comment (howardchung) [inline components/Player/PlayerName.jsx]: Note: We should try to reuse this component for rankings, peers, etc.

File: components/Table/PlayerMatchesTable.container.jsx [hunk 5]
```diff
@@ -29,7 +29,7 @@ class RequestLayer extends React.Component {
     this.props.getPlayerMatches(this.props.playerId, this.props.numMatches);
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentWillUpdate(nextProps) {
```
Comment (howardchung) [inline components/Table/PlayerMatchesTable.container.jsx]: What is the difference?

File: reducers/gotPlayer/player.js [hunk 6]
```diff
@@ -28,3 +31,19 @@ export default (state = initialState, action) => {
       return state;
   }
 };
+
+export const player = {
```
Comment (howardchung) [inline reducers/gotPlayer/player.js]: Why is this necessary?  Can't you just point the component to the path in the state?

File: .eslintrc.json (hunk 1 above)
Comment (mike-robertson) [inline .eslintrc.json]: lol yep

File: components/Table/PlayerMatchesTable.container.jsx (hunk 5 above)
Comment (mike-robertson) [inline components/Table/PlayerMatchesTable.container.jsx]: I think this is a preferred lifecycle method since I don't think it's called when the component is first loaded

File: reducers/gotPlayer/player.js (hunk 6 above)
Comment (mike-robertson) [inline reducers/gotPlayer/player.js]: Yes but I'd like to have better encapsulation of state so if we update the reducers we don't have to do any refactoring. Just trying this pattern out for now to see how it feels, cause it doesn't feel good that the containers need to know the structure of the store

File: components/Player/Player.jsx (hunk 3 above)
Comment (mike-robertson) [inline components/Player/Player.jsx]: withRouter just passes any router props down to it, it's just a different way of doing it. Not sure which is better but want to try to avoid passing props down a lot of components if we can avoid it (just learned about this method)

File: reducers/gotPlayer/player.js (hunk 6 above)
Comment (howardchung) [inline reducers/gotPlayer/player.js]: I don't like how we basically have to redeclare all of the props here.

If we add a prop in the API, it's a lot of code to add just to get it to show up in the UI.

With Jade we could start editing the template immediately and just reference the new prop.

Comment (mike-robertson) [inline reducers/gotPlayer/player.js]: if you change a prop you have to change it in every template as well

Comment (howardchung) [inline reducers/gotPlayer/player.js]: In this case we have to add it here and add/change the component.  That's two places
